cpp/daal/src/sycl/select_indexed.cpp

```diff
@@ -42,6 +45,13 @@ void run_quick_select_simd(ExecutionContextIface & context, ClKernelFactoryIface
                            uint32_t vectorSize, uint32_t lastVectorSize, uint32_t vectorOffset, QuickSelectIndexed::Result & result,
                            services::Status * status)
 {
+    if (nRndSeq > maxInt32 || nRndSeq == 0 || vectorSize > maxInt32 || lastVectorSize > vectorSize || nK > vectorSize || nK > lastVectorSize
+        || vectorOffset > maxInt32)
+    {
+        services::internal::tryAssignStatus(status, Status(services::ErrorBufferSizeIntegerOverflow));
```

Comment:
Are you sure that this status code is the best choice for every condition in 'if'? For instance, if nRndSeq == 0, is it really integer overflow for buffer size?